docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+        1. A cluster reaches one of the following end stages: `error`, `cancelled`, `installed`
+            - Each of these cluster stages represent an end state and that the cluster attempted installation.
+    - Sending data is non-blocking and sent as best effort
+3. Add logic for authenticating to the `DATA_COLLECTION_URL` 
+
+### Risks and Mitigations
+
+#### Risk 1: Overusing Customer's Network & CPU
+
+On-premises users might have large-scale clusters so network usage and processor usage could be impacted depending on how large the data set is.
+
+##### Mitigation
+
+To avoid user workflow interruption, data collection, processing, and sending can happen as a concurrent thread. Data sending should be non-blocking and best effort.
+
+#### Risk 2: Storage Limits and Data Loss
+
+Large-scale clusters most likely means large data sets that are sent to the event streaming service. There is a risk that amount of data could overflow the storage limits of the server used. 
+
+Furthermore, if the limits are reached, how will newer data be handled? Will the pre-existing data get overwritten or will the new data be rejected? There's potential for lost data here.
+
+##### Mitigation
+
+More research needs to be done here to see what the limits are and how we can overcome them. If using RH Ingress, there could be a daily job that would export data from here and import it into our own storage.
+
+#### Risk 3: DDoS-ing Insights Ingress Server
+
+Without any constraints, it's possible for the assisted service to send data to the data collection server repeatedly. This can prevent other clients from sending their data, and cause loss of data, or prevent us from reaching the data since the server will be busy with the repeated connections.
+
+##### Mitigation
+
+1. Clearly defining when events are sent (see `Data Sending Flow` under the `Proposal` section above)
+2. Data is sent as best effort
+3. Retry will be implemented as a future enhancement
+    - Not retrying until a timeout has been reached and there's no response from the server
+    - Each retry timeout will be longer (exponential backoff and jitter)
+    - Data will no longer be sent for this cluster once the maximum number of retries is hit
+4. Using a remote configuration (see section under `Proposal` above)
+
+#### Risk 4: Duplicate Data
+
+With all the cases for sending data, there's a high chance for duplicated data to be sent to the data collection server. It can skew the analysis of data and it won't be representative of the actual data from the field.
+
+##### Mitigation
+
+A field can be used as a marker to indicate the data has been sent to the event streaming service or that the data already exists. One option is to use the timestamp, cluster ID, and user email to detect duplicate events.
+
+## Design Details
+
+### Open Questions
+
+1. Is the remote configuration a good idea? A good amount of customers using this feature are Telcos and they would be disconnected
```

Comment:
In disconnected we're not going to get this data anyway right? Not sure how that relates to the remote config.

Reply:
That's right! The meeting helped clarified the end goal so this is not relevant at all. Thanks!